sig-segv on the a given mcpfile.yaml

Whereas before (and sorry I can't give you a version for "before") I had a working mcp server with launch as:
```bash
./genmcp run -f examples/netedge-tools/mcpfile.yaml
```

same yaml, but rebased to latest (as of 14:46 EST Oct 7, 2025), with reported version:
```bash
$ ./genmcp version
genmcp version development@5e2e7e4
```
I am now seeing on start:
```bash
$ ./genmcp run -f examples/netedge-tools/mcpfile.yaml
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x100692a14]

goroutine 1 [running]:
github.com/genmcp/gen-mcp/pkg/mcpfile.(*ServerRuntime).Validate(0x0)
	/Users/btofel/workspace/gen-mcp/pkg/mcpfile/validation.go:131 +0x24
github.com/genmcp/gen-mcp/pkg/mcpfile.(*MCPFile).Validate(0x1400023fc70, 0x103689600)
	/Users/btofel/workspace/gen-mcp/pkg/mcpfile/validation.go:22 +0x3e8
github.com/genmcp/gen-mcp/pkg/cli.executeRunCmd(0x14000159000?, {0x100a48542?, 0x4?, 0x100a48546?})
	/Users/btofel/workspace/gen-mcp/pkg/cli/run.go:52 +0xb4
github.com/spf13/cobra.(*Command).execute(0x103d49600, {0x140002535e0, 0x2, 0x2})
	/Users/btofel/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1019 +0x82c
github.com/spf13/cobra.(*Command).ExecuteC(0x103d49320)
	/Users/btofel/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1148 +0x384
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/btofel/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1071
github.com/genmcp/gen-mcp/pkg/cli.Execute({0x0?, 0x140000021c0?})
	/Users/btofel/workspace/gen-mcp/pkg/cli/root.go:25 +0x98
main.main()
	/Users/btofel/workspace/gen-mcp/cmd/genmcp/main.go:10 +0x2c
```

the mpcfile.yaml in question is (and again, sorry I don't have a minimal test case file here):
```yaml
mcpFileVersion: 0.0.1
servers:
- name: NETEDGE Network Diagnosis Tools
  version: 0.0.1
  runtime:
    streamableHttpConfig:
      port: 8080
    transportProtocol: streamablehttp
  tools:

  - name: inspect_route
    title: Inspect Route (detailed)
    description: |
      Retrieve the Route resource and attempt to inspect the associated Service and Endpoints.
      This tool will try to extract the service name from the Route and then fetch the
      Service and Endpoints objects. Requires `oc` CLI and either `jq` or `python3` available
      on the gen-mcp host.
    inputSchema:
      type: object
      properties:
        name:
          type: string
          description: Route name
        namespace:
          type: string
          description: Route namespace
      required:
      - name
      - namespace
    invocation:
      cli:
        command: |
          ns="{namespace}"
          name="{name}"
          tmpf=$(mktemp)
          oc get route "$name" -n "$ns" -o json > "$tmpf" || exit 1
          echo "=== ROUTE ==="
          cat "$tmpf"
          svc=""
          if command -v jq >/dev/null 2>&1; then
            svc=$(jq -r '.spec.to.name // ""' "$tmpf")
          fi
          if [ -n "$svc" ]; then
            echo "=== SERVICE: $svc ==="
            oc get svc "$svc" -n "$ns" -o json || true
            echo "=== ENDPOINTS: $svc ==="
            oc get endpoints "$svc" -n "$ns" -o json || true
          else
            echo "Could not extract service name from route. Service-specific inspection skipped."
          fi
  - name: get_service_endpoints
    title: Get Service Endpoints
    description: Return Endpoints object for a Service
    inputSchema:
      type: object
      properties:
        service:
          type: string
          description: Service name
        namespace:
          type: string
          description: Namespace of the service
      required:
      - service
      - namespace
    invocation:
      cli:
        command: "oc get endpoints {service} -n {namespace} -o json"

  - name: query_prometheus
    title: Query Prometheus (range)
    description: Run a Prometheus `query_range` query and return JSON. Requires `curl` on the gen-mcp host.
    inputSchema:
      type: object
      properties:
        prometheus_url:
          type: string
          description: Base URL for Prometheus (e.g. http://prometheus:9090)
        query:
          type: string
          description: PromQL expression
        start:
          type: string
          description: RFC3339 or unix timestamp for start
        end:
          type: string
          description: RFC3339 or unix timestamp for end
        step:
          type: string
          description: Step duration (e.g. 30s)
      required:
      - prometheus_url
      - query
      - start
      - end
      - step
    invocation:
      cli:
        command: |
          curl -sG "{prometheus_url}/api/v1/query_range" \
            --data-urlencode "query={query}" \
            --data-urlencode "start={start}" \
            --data-urlencode "end={end}" \
            --data-urlencode "step={step}"

  - name: get_coredns_config
    title: Get CoreDNS ConfigMap
    description: Retrieve a CoreDNS ConfigMap (or any ConfigMap) by name and namespace.
    inputSchema:
      type: object
      properties:
        configmap:
          type: string
          description: ConfigMap name
        namespace:
          type: string
          description: Namespace
      required:
      - configmap
      - namespace
    invocation:
      cli:
        command: "oc get configmap {configmap} -n {namespace} -o yaml"

  - name: probe_dns_local
    title: Probe DNS from gen-mcp host
    description: Run a DNS query using local `dig` or `nslookup` on the gen-mcp host.
    inputSchema:
      type: object
      properties:
        server:
          type: string
          description: DNS server (IP or host)
        name:
          type: string
          description: DNS name to query
        type:
          type: string
          description: DNS record type (A, AAAA, TXT, etc.)
      required:
      - server
      - name
      - type
    invocation:
      cli:
        command: |
          s="{server}"
          n="{name}"
          t="{type}"
          if command -v dig >/dev/null 2>&1; then
            dig @$s $n $t +noall +answer || true
          else
            nslookup -type=$t $n $s || echo "dig/nslookup not available"
          fi

  - name: exec_dns_in_pod
    title: Exec DNS probe in ephemeral pod
    description: Run a short-lived pod that performs a DNS query using an image containing DNS tools.
      The cluster must allow the image to be pulled from the container registry.
    inputSchema:
      type: object
      properties:
        namespace:
          type: string
          description: Namespace to run the ephemeral pod in
        server:
          type: string
          description: DNS server to query
        name:
          type: string
          description: DNS name to query
        type:
          type: string
          description: DNS record type
      required:
      - namespace
      - server
      - name
      - type
    invocation:
      cli:
        command: |
          oc run dnsquery --rm -i --restart=Never -n {namespace} --image=infoblox/dnstools --command -- sh -c "dig @{server} {name} {type}"
```

--- a/pkg/mcpfile/parser.go
+++ b/pkg/mcpfile/parser.go
@@ -50,6 +50,10 @@ func (m *MCPFile) UnmarshalJSON(data []byte) error {
 		}
 	}
 
+	if m.FileVersion != MCPFileVersion {
+		return fmt.Errorf("invalid mcp file version %s, expected %s - please migrate your file and handle any breaking changes", m.FileVersion, MCPFileVersion)
+	}
+
 	// Unmarshal the rest into MCPServer
 	if err := json.Unmarshal(data, &m.MCPServer); err != nil {
 		return err

--- a/pkg/mcpfile/parser_test.go
+++ b/pkg/mcpfile/parser_test.go
@@ -11,9 +11,10 @@ import (
 
 func TestParseMcpFile(t *testing.T) {
 	tt := map[string]struct {
-		testFileName string
-		expected     *MCPFile
-		wantErr      bool
+		testFileName  string
+		expected      *MCPFile
+		wantErr       bool
+		errorContains string
 	}{
 		"no servers": {
 			testFileName: "no-servers.yaml",
@@ -490,6 +491,11 @@ func TestParseMcpFile(t *testing.T) {
 				},
 			},
 		},
+		"invalid version 0.0.1": {
+			testFileName:  "invalid-file-version.yaml",
+			wantErr:       true,
+			errorContains: "invalid mcp file version",
+		},
 	}
 
 	for testName, testCase := range tt {
@@ -498,6 +504,7 @@ func TestParseMcpFile(t *testing.T) {
 			mcpFile, err := ParseMCPFile(fmt.Sprintf("./testdata/%s", testCase.testFileName))
 			if testCase.wantErr {
 				assert.Error(t, err, "parsing mcp file should cause an error")
+				assert.ErrorContains(t, err, testCase.errorContains, "the error should contain the right message")
 			} else {
 				assert.NoError(t, err, "parsing mcp file should succeed")
 			}

--- a/pkg/mcpfile/testdata/invalid-file-version.yaml
+++ b/pkg/mcpfile/testdata/invalid-file-version.yaml
@@ -1,0 +1,20 @@
+mcpFileVersion: 0.0.1
+servers:
+- name: test-server
+  version: 1.0.0
+  tools:
+  - name: get_user_by_company
+    title: Users Provider
+    description: Get list of users from a given company
+    inputSchema:
+      type: object
+      properties:
+        companyName:
+          type: string
+          description: Name of the company
+      required:
+      - companyName
+    invocation:
+      http:
+        url: http://localhost:5000
+        method: POST

